packages/boxel-ui/addon/components/dropdown-button/index.gts

```diff
@@ -0,0 +1,93 @@
+import type { TemplateOnlyComponent } from '@ember/component/template-only';
+import { type WithBoundArgs } from '@glint/template';
+import { hash } from '@ember/helper';
+import BoxelDropdown from '../dropdown';
+import BoxelMenu from '../menu';
+import cn from '@cardstack/boxel-ui/helpers/cn';
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+
+interface Signature {
+  Element: HTMLButtonElement;
+  Args: {
+    icon: string;
+    label: string;
+    class?: string;
+    contentClass?: string;
+    noHoverStyle?: boolean;
+    size?: number;
+    iconSize?: number;
+  };
+  Blocks: {
+    default: [
+      {
+        Menu: WithBoundArgs<typeof BoxelMenu, 'closeMenu'>;
+        close: () => void;
+      },
+    ];
+  };
+}
+
+const DropdownButton: TemplateOnlyComponent<Signature> = <template>
+  <BoxelDropdown @contentClass={{@contentClass}}>
+    <:trigger as |bindings|>
+      <button
+        {{bindings}}
+        class={{cn
+          'boxel-dropdown-button'
+          'boxel-dropdown-button__reset'
+          'boxel-dropdown-button__trigger'
+          @class
+        }}
+        aria-label={{@label}}
+        data-test-boxel-dropdown-button
+        ...attributes
+      >
+        {{#if @icon}}
+          {{svgJar
+            @icon
+            width=(if @iconSize @iconSize 16)
+            height=(if @iconSize @iconSize 16)
+          }}
+        {{/if}}
+      </button>
+    </:trigger>
+    <:content as |dd|>
+      {{yield
+        (hash Menu=(component BoxelMenu closeMenu=dd.close) close=dd.close)
+      }}
+    </:content>
+  </BoxelDropdown>
+  <style>
+    @layer {
```

Comment:
What is the advantage of using @layer over adding a custom class to the dropdown and making rules depending on that?

Reply:
`@layer` makes sure that if the custom class css and preexisting component css are conflicting, the custom class overrides the component css